Reproducible and Open Data Science - Training Course - RODS17
=============================================================

Description 
-------------

In an age of increasingly complex and data-intensive, collaborative scientific practices, scandals of irreproducibility, and a growing societal ethos of transparency and accountability, a new paradigm has arisen: Open Science. In this three day course, we will introduce to you the three organizing principles and practices that undergird this paradigm:
- Open Access scholarly publishing
- Open Source software development
- Open Data integration and sharing

To support each of these, numerous tools exist, which we will introduce to you. The examples are going to be based on bioinformatics, but the principles are broadly applicable. You do not need to have any particular programming or otherwise computational experience beyond what is normally required of a scientist in graduate school and beyond, i.e., you should not be afraid of the computer and be able to edit text files.

1. Introduction: What is Open Science?
2. 
3. 
4. 
5. 
6. 
7.
8. 
9. 
10.
11.
12.

Mon, April 17th

    09:30 - 10:00 Introduction to the course and self presentation of the participants
    10:00 - 11:00 Possibilities and limitations of NGS sequencing technologies. Choose adequate sequencing for your biological question
    11:00 - 11:30 Coffee Break
    11:30 - 12:30 Steps in the analysis of RNA-Seq differential expression experiments
    12:30 - 14:00 LUNCH BREAK
    14:00 - 16:00 Interpret what are fastq files and what is their content. Use software like FastQC to process fastq files and produce QC reports. Read QC reports of raw data to assess the general quality of data and presence of sequence bias. Use tools such as seqtk, cutadapt and trimmomatic to remove low quality bases, adaptors and other artefactual sequences from your reads.
    16:00 - 16:30 Tea Break
    16:30 - 18:00 What is a reference genome, versioning and where to obtain genomes. Alignment software: hisat2; bwa; salmon. Run an alignment: the SAM/BAM alignment format.

Tue, April 18th

    09:30 - 10:00 Morning Wrap-up (what have we done so far?)
    10:00 - 11:00 Visualizing alignments in IGV for single genes.
    11:00 - 11:30 Coffee Break
    11:30 - 12:30 Use tools such as RSeQC and Qualimap to assess quality of alignments
    12:30 - 14:00 LUNCH BREAK
    14:00 - 16:00 What is a reference gene annotation, versioning and where to obtain. The process of generating gene counts from genome aligments. Use tools such as htseq-counts to generate table of gene counts.
    16:00 - 16:30 Tea Break
    16:30 - 18:00 Using the R package edgeR and DESeq2 in Galaxy to produce a pairwise differential expression analysis

Wed, April 19th

    09:30 - 10:00 Morning Wrap-up (what have we done so far?)
    10:00 - 11:00 Use edgeR and DESeq2 in R and RStudio.
    11:00 - 11:30 Coffee Break
    11:30 - 12:30 Interpretation and visualization of results.
    12:30 - 14:00 LUNCH BREAK
    14:00 - 16:00 Interpretation and visualization of results.
    16:00 - 16:30 Tea Break
    16:30 - 18:00 Use more complex settings: Generalized Linear Models

Thu, April 20th

    09:30 - 10:00 Morning wrap-up (what have we done so far?)
    10:00 - 11:00 Use more complex settings: Generalized Linear Models
    11:00 - 11:30 Coffee Break
    11:30 - 12:30 Functional annotations: what are these and where to get them. The statistics behind functional enrichment analysis.
    12:30 - 14:00 LUNCH BREAK
    14:00 - 16:00 Using functional enrichment analysis with your lists of genes.
    16:00 - 16:30 Tea Break
    16:30 - 18:00 Final wrap-up Session.

