
<!DOCTYPE HTML>
<html lang="" >
    <head>
        <meta charset="UTF-8">
        <meta content="text/html; charset=utf-8" http-equiv="Content-Type">
        <title>How to be formally explicit about data and concepts Â· GitBook</title>
        <meta http-equiv="X-UA-Compatible" content="IE=edge" />
        <meta name="description" content="">
        <meta name="generator" content="GitBook 3.2.3">
        
        
        
    
    <link rel="stylesheet" href="../gitbook/style.css">

    
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-quizzes/quizzes.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-image-captions/image-captions.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-highlight/website.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-search/search.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-fontsettings/website.css">
                
            
        

    

    
        
    
        
    
        
    
        
    
        
    
        
    

        
    
    
    <meta name="HandheldFriendly" content="true"/>
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <link rel="apple-touch-icon-precomposed" sizes="152x152" href="../gitbook/images/apple-touch-icon-precomposed-152.png">
    <link rel="shortcut icon" href="../gitbook/images/favicon.ico" type="image/x-icon">

    
    <link rel="next" href="../SCIENTIFIC_SOFTWARE/" />
    
    
    <link rel="prev" href="../WORKFLOWS/" />
    

    </head>
    <body>
        
<div class="book">
    <div class="book-summary">
        
            
<div id="book-search-input" role="search">
    <input type="text" placeholder="Type to search" />
</div>

            
                <nav role="navigation">
                


<ul class="summary">
    
    

    

    
        
        
    
        <li class="chapter " data-level="1.1" data-path="../">
            
                <a href="../">
            
                    
                    Front matter
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.2" data-path="../ROADMAP/">
            
                <a href="../ROADMAP/">
            
                    
                    Who is this for?
            
                </a>
            

            
        </li>
    

    
        
        <li class="header">Open Science</li>
        
        
    
        <li class="chapter " data-level="2.1" data-path="../LITERATURE_STUDY/">
            
                <a href="../LITERATURE_STUDY/">
            
                    
                    How to navigate the scientific record
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="2.2" data-path="../VERSIONING/">
            
                <a href="../VERSIONING/">
            
                    
                    How to cope with evolving research outputs
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="2.3" data-path="../REPRODUCIBILITY/">
            
                <a href="../REPRODUCIBILITY/">
            
                    
                    How to make your research reproducible
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="2.4" data-path="../PUBLISHING/">
            
                <a href="../PUBLISHING/">
            
                    
                    How to publish your research with impact
            
                </a>
            

            
        </li>
    

    
        
        <li class="header">Open Data</li>
        
        
    
        <li class="chapter " data-level="3.1" data-path="../DATA_CAPTURE/">
            
                <a href="../DATA_CAPTURE/">
            
                    
                    How to capture data to produce reliable records
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="3.2" data-path="../DATA_MANAGEMENT/">
            
                <a href="../DATA_MANAGEMENT/">
            
                    
                    How to manage data and plan for it
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="3.3" data-path="../DATA_SHARING/">
            
                <a href="../DATA_SHARING/">
            
                    
                    How to share research data fairly and sensibly
            
                </a>
            

            
        </li>
    

    
        
        <li class="header">Open Source</li>
        
        
    
        <li class="chapter " data-level="4.1" data-path="../WORKFLOWS/">
            
                <a href="../WORKFLOWS/">
            
                    
                    How to do analyses in a workflow-oriented manner
            
                </a>
            

            
        </li>
    
        <li class="chapter active" data-level="4.2" data-path="./">
            
                <a href="./">
            
                    
                    How to be formally explicit about data and concepts
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="4.3" data-path="../SCIENTIFIC_SOFTWARE/">
            
                <a href="../SCIENTIFIC_SOFTWARE/">
            
                    
                    How to improve scientific source code development
            
                </a>
            

            
        </li>
    

    
        
        <li class="divider"></li>
        
        
    
        <li class="chapter " data-level="5.1" data-path="../REFERENCES/">
            
                <a href="../REFERENCES/">
            
                    
                    References
            
                </a>
            

            
        </li>
    

    

    <li class="divider"></li>

    <li>
        <a href="https://www.gitbook.com" target="blank" class="gitbook-link">
            Published with GitBook
        </a>
    </li>
</ul>


                </nav>
            
        
    </div>

    <div class="book-body">
        
            <div class="body-inner">
                
                    

<div class="book-header" role="navigation">
    

    <!-- Title -->
    <h1>
        <i class="fa fa-circle-o-notch fa-spin"></i>
        <a href=".." >How to be formally explicit about data and concepts</a>
    </h1>
</div>




                    <div class="page-wrapper" tabindex="-1" role="main">
                        <div class="page-inner">
                            
<div id="book-search-results">
    <div class="search-noresults">
    
                                <section class="normal markdown-section">
                                
                                <h1 id="how-to-be-formally-explicit-about-data-and-concepts">How to be formally explicit about data and concepts</h1>
<p>In a number of locations in these pages we refer to cases where there is a common understanding among stakeholders of the meaning of certain terms. For example, in discussing <a href="../SCIENTIFIC_SOFTWARE/#community-conventions">community conventions in open source software development</a>, certain specific keywords are used in package names and metadata files describing packages; in the issuing of <a href="../VERSIONING/#version-numbers">version numbers</a> for software, where the different parts of a version number contain embedded meaning about how one version differs from the next; in literature search, where search terms might be anchored to specific interpretations, as in the case of <a href="../LITERATURE_STUDY/#medical-subject-headings-mesh">MeSH terms</a>; in the description of experiments, where <a href="../DATA_CAPTURE/#minimal-information-standards">minimal information standards</a> define checklists of what to report about an assay. These are all instances where a community has defined the <a href="https://en.wikipedia.org/wiki/Semantics" target="_blank">semantics</a> of terms, usually for the purpose of integrating information (articles, software) from different sources. In an open world where we want to share our research output with others and others to use our data in turn, similar challenges (and solutions) arise when integrating research data. Here we will discuss some of the general principles to guide is in making our data more amenable to integration.</p>
<h2 id="how-to-structure-data">How to structure data</h2>
<p>Sometimes your data is <a href="../DATA_CAPTURE/">captured</a> by a machine in such a way that you might never edit it &quot;by hand&quot; and it is automatically structured according to community standards (such as <a href="../REFERENCES/#Cock2009">FASTQ</a> in the case of high-throughput DNA sequencing). In other cases, you might collect and collate data by the hand, for example in the field, or while scanning through the literature or perusing online resources. A common approach is then to enter data in a spreadsheet. In this, there are certainly tidy and untidy ways to do this.</p>
<p><a name="s1"></a>
<img src="journal.pcbi.1005097.g001.PNG" alt="Example 1, untidy versus tidy data"></p>
<p>Consider <a href="#s1">example 1</a>. <em>Dataset A is untidy because it mixes observational units (species, location of observations, measurements about individuals), the units are mixed and listed with the observations, more than one variable is listed (both latitude and longitude for the coordinates, and genus and species for the species names), and several formats are used in the same column for dates and geographic coordinates. Dataset B is an example of a tidy version of dataset A that reduces the amount of information that is duplicated in each row, limiting chances of introducing mistakes in the data. By having species in a separate table, they can be identified uniquely using the Taxonomic Serial Number (TSN) from the Integrated Taxonomic Information System (ITIS), and it makes it easy to add information about the classification of these species. It also allows researchers to edit the taxonomic information independently from the table that holds the measurements about the individuals. Unique values for each observational unit facilitate the programmatic combination of information using &quot;join&quot; operations. With this example, if the focus of the study for which these data were collected is based upon the size measurements of the individuals (weight and length), information about &quot;where&quot;, &quot;when&quot;, and &quot;what&quot; animals were measured can be considered metadata. Using the tidy format makes this distinction clearer.</em> (reproduced from [<a href="../REFERENCES/#Hart2016">Hart2016</a>])</p>
<p>Assuming you follow the advice from Hart et al. faithfully, your next decision will be what file format to store your data in. Although you might be very productive in a spreadsheet editor such as Microsoft Excel, resist the temptation to store your data in Excel&apos;s proprietary file format (<code>.xls</code> or <code>.xlsx</code>). Not only is Excel format readable by fewer programs than a plain text format such as <a href="https://en.wikipedia.org/wiki/Tab-separated_values" target="_blank">tab-separated values</a>, which reduces the ways in which the data can be re-used without further conversion, the risk that your data will be unintentionally altered (corrupted) by Excel is very real: <a href="../REFERENCES/#Ziemann2016">gene name errors (because of this) are widespread in the scientific literature</a>.</p>
<p>The point of structuring your data in &quot;tidy&quot; form as tabular, plain text will be immediately obvious once you (or your collaborators, or people using your data) try to analyse the data by including it, for example, in a <a href="../WORKFLOWS/">computational analysis workflow</a>: software such as the R environment for statistical computing, will simply refuse to read untidy data correctly, and your chosen platform for data <a href="../VERSIONING/">versioning</a> or <a href="../DATA_SHARING/">sharing</a> will likely also not display the data correctly.</p>
<h2 id="but-what-does-it-all-mean">But what does it all mean?</h2>
<p>The advice to tidy your data and store it in an open file format is good, but it is incomplete. Whereas the change from <code>weight</code> and <code>length</code> to <code>weight_kg</code> and <code>length_cm</code> is an improvement, the encoding of the latitude and longitude coordinates in the &quot;tidy&quot; version still omits crucial information: geographic coordinates can be expressed in different units (e.g. in minutes and seconds, or in decimal values), and it is not made explicit which one is used, i.e. the semantics are still somewhat unclear. To address this, we might pick something like <code>longitude_decimal</code> and <code>latitude_decimal</code>, but this, like the other column headers, is actually an arbitrary choice: maybe my colleague uses <code>decimal_longitude</code> or <code>longitude_dec</code>, which would be a nightmare if we try to merge our tables automatically in a script that is not programmed to &quot;know&quot; that these are equivalent. We need to agree on a common terminology for this.</p>
<p>Common terminologies expressed as structured data in open file formats are called <a href="https://en.wikipedia.org/wiki/Ontology_(information_science" target="_blank">ontologies</a>) and <a href="https://en.wikipedia.org/wiki/Controlled_vocabulary" target="_blank">controlled vocabularies</a>. In the life sciences, numerous of these ontologies and controlled vocabularies have been and are being developed for the jargon of different subdomains. For example, a suitable controlled vocabulary for the terms involved in recording <a href="http://www.gbif.org/occurrence" target="_blank">species occurrences</a> is the <a href="http://rs.tdwg.org/dwc" target="_blank">DarwinCore</a>, which has terms for geographical coordinates in decimal form. If we choose to use these terms, we specify the vocabulary that they came from by a prefix (<code>dwc:</code>, which identifies the DarwinCore vocabulary), so that the column headers then become <a href="https://terms.tdwg.org/wiki/dwc:decimalLongitude" target="_blank">dwc:decimalLongitude</a> and <a href="https://terms.tdwg.org/wiki/dwc:decimalLatitude" target="_blank">dwc:decimalLatitude</a>.</p>
<h2 id="useful-ontologies-and-controlled-vocabularies">Useful ontologies and controlled vocabularies</h2>
<p>Many databases have started to adopt terms from ontologies with which to annotate records. We have already seen how databases with species occurrences such as <a href="http://www.gbif.org/occurrence" target="_blank">GBIF</a> uses <a href="http://rs.tdwg.org/dwc" target="_blank">DarwinCore</a> terms. Likewise, GenBank uses the <a href="http://www.sequenceontology.org/" target="_blank">Sequence Ontology</a> as the terminology for sequence features such as the <a href="http://www.sequenceontology.org/so_wiki/index.php/Category:SO:0000316_%21_CDS" target="_blank">CDS</a>, and terms from the Gene Ontology for gene functions, such as GO:<a href="http://amigo.geneontology.org/amigo/term/GO:0006118" target="_blank">0006118</a> <em>electron transport</em> and GO:<a href="http://amigo.geneontology.org/amigo/term/GO:0005507" target="_blank">0005507</a> <em>copper ion binding</em>, in the record below. (Note, also, the <a href="../VERSIONING/#versioning-public-database-records">versioned</a> accession number <code>BT022039.1</code> and the way in which the taxon is anchored to an online taxonomy as taxon:<a href="http://ncbi.nlm.nih.gov/taxonomy/3702" target="_blank">3702</a> along similar lines as was done in the &quot;tidy&quot; data example.)</p>
<pre><code>    LOCUS       BT022039                 536 bp    mRNA    linear   PLN 03-MAY-2005
    DEFINITION  Arabidopsis thaliana At1g22480 gene, complete cds.
    ACCESSION   BT022039
    VERSION     BT022039.1
    KEYWORDS    FLI_CDNA.
    SOURCE      Arabidopsis thaliana (thale cress)
      ORGANISM  Arabidopsis thaliana
                Eukaryota; Viridiplantae; Streptophyta; Embryophyta; Tracheophyta;
                Spermatophyta; Magnoliophyta; eudicotyledons; Gunneridae;
                Pentapetalae; rosids; malvids; Brassicales; Brassicaceae;
                Camelineae; Arabidopsis.
    REFERENCE   1  (bases 1 to 536)
      AUTHORS   Cheuk,R., Chen,H., Kim,C.J., Shinn,P. and Ecker,J.R.
      TITLE     Arabidopsis ORF clones
      JOURNAL   Unpublished
    REFERENCE   2  (bases 1 to 536)
      AUTHORS   Cheuk,R., Chen,H., Kim,C.J., Shinn,P. and Ecker,J.R.
      TITLE     Direct Submission
      JOURNAL   Submitted (03-MAY-2005) Salk Institute Genomic Analysis Laboratory
                (SIGnAL), Plant Biology Laboratory, The Salk Institute for
                Biological Studies, 10010 N. Torrey Pines Road, La Jolla, CA 92037,
                USA
    FEATURES             Location/Qualifiers
         source          1..536
                         /organism=&quot;Arabidopsis thaliana&quot;
                         /mol_type=&quot;mRNA&quot;
                         /db_xref=&quot;taxon:3702&quot;
                         /chromosome=&quot;1&quot;
                         /clone=&quot;S80220&quot;
                         /ecotype=&quot;Columbia&quot;
                         /note=&quot;This clone is in pUNI 51&quot;
         CDS             6..530
                         /note=&quot;plastocyanin-like domain-containing protein
                         go_function: copper ion binding [goid 0005507];
                         go_process: electron transport [goid 0006118]&quot;
                         /codon_start=1
                         /product=&quot;At1g22480&quot;
                         /protein_id=&quot;AAY25451.1&quot;
                         /translation=&quot;MSTLLGCLVLIFSMVAQASSASLTVNWSLGTDYTPLTTGKTFSV
                         GDTIVFNYGAGHTVDEVSENDYKSCTLGNSITSDSSGTTTIALTTTGPRYFICGIPGH
                         CAAGMKLAVTVASNSSNGVAGGTTTPTPFTGGGGGYNPTTTQAIPCAAWAVSCPLRAL
                         VATWAVVFYALALS&quot;
    ORIGIN      
            1 aaaatatgag cacacttctt ggttgtcttg tcctcatatt ctctatggtc gcacaggcct
           61 catccgccag tcttacggtg aactggtccc ttggcaccga ctacactccg ctcaccactg
          121 gaaagacctt ctctgtcggc gataccatag tgttcaatta tggtgcgggt cacacggtgg
          181 atgaagtgag cgagaacgac tacaagagtt gcactctagg gaactccatt acgtccgaca
          241 gcagcggaac cacgaccata gctctcacga ccactggtcc tcgctacttc atctgtggaa
          301 tccccggcca ttgcgctgcc ggtatgaagc tcgcagtcac cgtcgcgtcg aactcttcaa
          361 acggtgtagc tggtggcacc actacaccaa ccccattcac cggaggtggt ggtggttaca
          421 atcccacgac cacacaggcc attccttgtg cggcttgggc cgtgtcctgt ccattacggg
          481 ctttggttgc tacttgggcc gttgtttttt atgctttggc tttgtcttag ttgaaa
    //
</code></pre><p>As we have seen, ontology terms can be embedded in data files in a variety of ways. Beyond the examples shown here, there are also <a href="https://en.wikipedia.org/wiki/Resource_Description_Framework#Serialization_formats" target="_blank">open data formats</a> that are explicitly designed for linking ontology terms (and data values) together in <a href="https://www.w3.org/TR/2004/REC-rdf-concepts-20040210/#section-triples" target="_blank">statements</a> about the world. This is a technique that is intended to link distributed data together on the &quot;<a href="https://en.wikipedia.org/wiki/Semantic_Web" target="_blank">semantic web</a>&quot;, which, although very powerful, is not necessarily something you have to aim for if your goal is simply to adopt common terminology within existing data files. Irrespective of your specific reasons for wanting to use ontology terms, the <a href="https://bioportal.bioontology.org/" target="_blank">growing number of ontologies</a> complicates choosing the right one. It may therefore be useful to read the guidelines by [<a href="../REFERENCES/#Malone2016">Malone2016</a>] for this.</p>
<h2 id="expected-outcomes">Expected outcomes</h2>
<p>In this section we have discussed some of the pitfalls in assembling tidy data sets that can be shared and integrated with others. You should now:</p>
<ul>
<li>Be persuaded to tidily structure your data</li>
<li>Know to use open, simple data formats</li>
<li>Understand the purpose of ontologies and controlled vocabularies</li>
<li>Know some common techniques for how ontology terms are included in data files</li>
</ul>

                                
                                </section>
                            
    </div>
    <div class="search-results">
        <div class="has-results">
            
            <h1 class="search-results-title"><span class='search-results-count'></span> results matching "<span class='search-query'></span>"</h1>
            <ul class="search-results-list"></ul>
            
        </div>
        <div class="no-results">
            
            <h1 class="search-results-title">No results matching "<span class='search-query'></span>"</h1>
            
        </div>
    </div>
</div>

                        </div>
                    </div>
                
            </div>

            
                
                <a href="../WORKFLOWS/" class="navigation navigation-prev " aria-label="Previous page: How to do analyses in a workflow-oriented manner">
                    <i class="fa fa-angle-left"></i>
                </a>
                
                
                <a href="../SCIENTIFIC_SOFTWARE/" class="navigation navigation-next " aria-label="Next page: How to improve scientific source code development">
                    <i class="fa fa-angle-right"></i>
                </a>
                
            
        
    </div>

    <script>
        var gitbook = gitbook || [];
        gitbook.push(function() {
            gitbook.page.hasChanged({"page":{"title":"How to be formally explicit about data and concepts","level":"4.2","depth":1,"next":{"title":"How to improve scientific source code development","level":"4.3","depth":1,"path":"SCIENTIFIC_SOFTWARE/README.md","ref":"SCIENTIFIC_SOFTWARE/README.md","articles":[]},"previous":{"title":"How to do analyses in a workflow-oriented manner","level":"4.1","depth":1,"path":"WORKFLOWS/README.md","ref":"WORKFLOWS/README.md","articles":[]},"dir":"ltr"},"config":{"plugins":["mathjax","quizzes","codeblock-disable-glossary","image-captions"],"root":"./docs","styles":{"website":"styles/website.css","pdf":"styles/pdf.css","epub":"styles/epub.css","mobi":"styles/mobi.css","ebook":"styles/ebook.css","print":"styles/print.css"},"pluginsConfig":{"search":{},"autocover":{"title":"Open Science, Open Data, Open Source","author":"Pedro L. Fernandes and Rutger A. Vos","font":{"size":null,"family":"Impact","color":"#FFF"},"size":{"w":1800,"h":2360},"background":{"color":"#006600"}},"lunr":{"maxIndexSize":1000000,"ignoreSpecialCharacters":false},"codeblock-disable-glossary":{},"fontsettings":{"theme":"white","family":"sans","size":2},"quizzes":{},"highlight":{},"mathjax":{"forceSVG":false,"version":"2.6-latest"},"sharing":{"facebook":true,"twitter":true,"google":false,"weibo":false,"instapaper":false,"vk":false,"all":["facebook","google","twitter","weibo","instapaper"]},"theme-default":{"styles":{"website":"styles/website.css","pdf":"styles/pdf.css","epub":"styles/epub.css","mobi":"styles/mobi.css","ebook":"styles/ebook.css","print":"styles/print.css"},"showLevel":false},"image-captions":{"variable_name":"_pictures"}},"theme":"default","pdf":{"pageNumbers":true,"fontSize":12,"fontFamily":"Arial","paperSize":"a4","chapterMark":"pagebreak","pageBreaksBefore":"/","margin":{"right":62,"left":62,"top":56,"bottom":56}},"structure":{"langs":"LANGS.md","readme":"README.md","glossary":"GLOSSARY.md","summary":"SUMMARY.md"},"variables":{"_pictures":[{"backlink":"LITERATURE_STUDY/index.html#fig2.1.1","level":"2.1","list_caption":"Figure: Randall Munroe on open access publishing","alt":"Randall Munroe on open access publishing","nro":1,"url":"LS2.jpg","index":1,"caption_template":"Figure: _CAPTION_","label":"Randall Munroe on open access publishing","attributes":{},"skip":false,"key":"2.1.1"},{"backlink":"PUBLISHING/index.html#fig2.4.1","level":"2.4","list_caption":"Figure: Preprints","alt":"Preprints","nro":2,"url":"https://img.youtube.com/vi/2zMgY8Dx9co/0.jpg","index":1,"caption_template":"Figure: _CAPTION_","label":"Preprints","attributes":{},"skip":false,"key":"2.4.1"}]},"gitbook":"*"},"file":{"path":"SEMANTICS/README.md","mtime":"2017-10-02T14:11:52.000Z","type":"markdown"},"gitbook":{"version":"3.2.3","time":"2017-10-02T15:00:52.413Z"},"basePath":"..","book":{"language":""}});
        });
    </script>
</div>

        
    <script src="../gitbook/gitbook.js"></script>
    <script src="../gitbook/theme.js"></script>
    
        
        <script src="https://cdn.mathjax.org/mathjax/2.6-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-mathjax/plugin.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-quizzes/quizzes.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-codeblock-disable-glossary/disable-glossary.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-search/search-engine.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-search/search.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-lunr/lunr.min.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-lunr/search-lunr.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-sharing/buttons.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-fontsettings/fontsettings.js"></script>
        
    

    </body>
</html>

