
<!DOCTYPE HTML>
<html lang="" >
    <head>
        <meta charset="UTF-8">
        <meta content="text/html; charset=utf-8" http-equiv="Content-Type">
        <title>How to share research data fairly and sensibly Â· GitBook</title>
        <meta http-equiv="X-UA-Compatible" content="IE=edge" />
        <meta name="description" content="">
        <meta name="generator" content="GitBook 3.2.3">
        
        
        
    
    <link rel="stylesheet" href="../gitbook/style.css">

    
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-quizzes/quizzes.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-image-captions/image-captions.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-highlight/website.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-search/search.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-fontsettings/website.css">
                
            
        

    

    
        
    
        
    
        
    
        
    
        
    
        
    

        
    
    
    <meta name="HandheldFriendly" content="true"/>
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <link rel="apple-touch-icon-precomposed" sizes="152x152" href="../gitbook/images/apple-touch-icon-precomposed-152.png">
    <link rel="shortcut icon" href="../gitbook/images/favicon.ico" type="image/x-icon">

    
    <link rel="next" href="../WORKFLOWS/" />
    
    
    <link rel="prev" href="../DATA_MANAGEMENT/" />
    

    </head>
    <body>
        
<div class="book">
    <div class="book-summary">
        
            
<div id="book-search-input" role="search">
    <input type="text" placeholder="Type to search" />
</div>

            
                <nav role="navigation">
                


<ul class="summary">
    
    

    

    
        
        
    
        <li class="chapter " data-level="1.1" data-path="../">
            
                <a href="../">
            
                    
                    Front matter
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.2" data-path="../ROADMAP/">
            
                <a href="../ROADMAP/">
            
                    
                    Who is this for?
            
                </a>
            

            
        </li>
    

    
        
        <li class="header">Open Science</li>
        
        
    
        <li class="chapter " data-level="2.1" data-path="../LITERATURE_STUDY/">
            
                <a href="../LITERATURE_STUDY/">
            
                    
                    How to navigate the scientific record
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="2.2" data-path="../VERSIONING/">
            
                <a href="../VERSIONING/">
            
                    
                    How to cope with evolving research outputs
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="2.3" data-path="../REPRODUCIBILITY/">
            
                <a href="../REPRODUCIBILITY/">
            
                    
                    How to make your research reproducible
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="2.4" data-path="../PUBLISHING/">
            
                <a href="../PUBLISHING/">
            
                    
                    How to publish your research with impact
            
                </a>
            

            
        </li>
    

    
        
        <li class="header">Open Data</li>
        
        
    
        <li class="chapter " data-level="3.1" data-path="../DATA_CAPTURE/">
            
                <a href="../DATA_CAPTURE/">
            
                    
                    How to capture data to produce reliable records
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="3.2" data-path="../DATA_MANAGEMENT/">
            
                <a href="../DATA_MANAGEMENT/">
            
                    
                    How to manage data and plan for it
            
                </a>
            

            
        </li>
    
        <li class="chapter active" data-level="3.3" data-path="./">
            
                <a href="./">
            
                    
                    How to share research data fairly and sensibly
            
                </a>
            

            
        </li>
    

    
        
        <li class="header">Open Source</li>
        
        
    
        <li class="chapter " data-level="4.1" data-path="../WORKFLOWS/">
            
                <a href="../WORKFLOWS/">
            
                    
                    How to do analyses in a workflow-oriented manner
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="4.2" data-path="../SEMANTICS/">
            
                <a href="../SEMANTICS/">
            
                    
                    How to be formally explicit about data and concepts
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="4.3" data-path="../SCIENTIFIC_SOFTWARE/">
            
                <a href="../SCIENTIFIC_SOFTWARE/">
            
                    
                    How to improve scientific source code development
            
                </a>
            

            
        </li>
    

    
        
        <li class="divider"></li>
        
        
    
        <li class="chapter " data-level="5.1" data-path="../REFERENCES/">
            
                <a href="../REFERENCES/">
            
                    
                    References
            
                </a>
            

            
        </li>
    

    

    <li class="divider"></li>

    <li>
        <a href="https://www.gitbook.com" target="blank" class="gitbook-link">
            Published with GitBook
        </a>
    </li>
</ul>


                </nav>
            
        
    </div>

    <div class="book-body">
        
            <div class="body-inner">
                
                    

<div class="book-header" role="navigation">
    

    <!-- Title -->
    <h1>
        <i class="fa fa-circle-o-notch fa-spin"></i>
        <a href=".." >How to share research data fairly and sensibly</a>
    </h1>
</div>




                    <div class="page-wrapper" tabindex="-1" role="main">
                        <div class="page-inner">
                            
<div id="book-search-results">
    <div class="search-noresults">
    
                                <section class="normal markdown-section">
                                
                                <h1 id="how-to-share-research-data-fairly-and-sensibly">How to share research data fairly and sensibly</h1>
<p>Collaboration is a natural, essential element in research. However, sharing of resources amongst scientists should be a lot easier than it is, as it finds expected and unexpected barriers everywhere. In recent years, issues surrounding scientific data sharing have received a lot of attention (e.g. see [<a href="../REFERENCES#Gewin2016">Gewin2016</a>]), and this has led both to a better understanding of the right principles and practices that should surround such sharing as well as to better infrastructure. </p>
<h2 id="the-fair-guiding-principles">The FAIR Guiding Principles</h2>
<p>The principles that (should) guide scientific data sharing are abbreviated as <strong>FAIR</strong>, which stands for <em>Findable, Accessible, Interoperable, Reusable</em>. What is meant by this is outlined below, and discussed in much greater detail in [<a href="../REFERENCES#Wilkinson2016">Wilkinson2016</a>].</p>
<blockquote>
<p>To be <strong>Findable</strong>:</p>
<ul>
<li>F1. (meta)data are assigned a <a href="../VERSIONING">globally unique and persistent identifier</a></li>
<li>F2. data are described with rich metadata (defined by R1 below)</li>
<li>F3. metadata clearly and explicitly include the identifier of the data it describes</li>
<li>F4. (meta)data are registered or indexed in a searchable resource</li>
</ul>
<p>To be <strong>Accessible</strong>:</p>
<ul>
<li>A1. (meta)data are retrievable by their identifier using a standardised communications protocol</li>
<li>A1.1 the protocol is open, free, and universally implementable</li>
<li>A1.2 the protocol allows for an authentication and authorisation procedure, where necessary</li>
<li>A2. metadata are accessible, even when the data are no longer available</li>
</ul>
<p>To be <strong>Interoperable</strong>:</p>
<ul>
<li>I1. (meta)data use a formal, accessible, shared, and broadly applicable language for knowledge representation.</li>
<li>I2. (meta)data use <a href="../SEMANTICS">vocabularies</a> that follow FAIR principles</li>
<li>I3. (meta)data include qualified references to other (meta)data</li>
</ul>
<p>To be <strong>Reusable</strong>:</p>
<ul>
<li>R1. meta(data) are richly described with a plurality of accurate and relevant attributes</li>
<li>R1.1. (meta)data are released with a clear and accessible data usage license</li>
<li>R1.2. (meta)data are associated with detailed provenance</li>
<li>R1.3. (meta)data meet domain-relevant community standards</li>
</ul>
</blockquote>
<p>Many different standards, databases, and policies can be adopted and combined to develop practices that comply with these general principles. These are collected on <a href="https://fairsharing.org/" target="_blank">FAIRSharing</a>, which also contains an extensive educational section. </p>
<p>Assuming that we are persuaded by the wisdom of these FAIR principles, we may want to adopt them in our own research when it comes to sharing our data in various contexts. Among these contexts we can at least distinguish between the sharing of raw and intermediate data (for example, within a collaborative network) and the publishing of our &quot;final&quot;, result data, i.e. when we finished with our analyses and want to share our conclusions with the world. </p>
<p>In either case, many of the FAIR principles can be implemented following guidelines we establish elsewhere on these pages. For example, to be properly <strong>Findable</strong>, our data should treated at least according to our suggestions in the section on <a href="../VERSIONING">versioning</a>; and to be <strong>Interoperable</strong> and <strong>Reusable</strong> we should describe our data collection process following reporting standards such as described in the section on <a href="../DATA_CAPTURE">data capture</a>, we should express the meaning of our data and metadata using clear, unambiguous <a href="../SEMANTICS">semantics</a>, and we should adopt open, community standards for representing our data, as we elaborate on in the section on <a href="../SCIENTIFIC_SOFTWARE">open source</a>. </p>
<p>What remains to be discussed are the options for making our data <strong>Accessible</strong>, and for this there are different technologies to choose from depending on whether we are sharing raw and intermediate data or whether we are publishing result data. In the following sections we will discuss these options.</p>
<h2 id="sharing-raw-and-intermediate-data">Sharing raw and intermediate data</h2>
<p>In the more private context of a research collaboration before publication it might seem that it does not matter that much how data are shared: the collaborative network is probably small enough that most people know each other and can jointly reconstruct where the data are, what has been done to it so far, and so on. However, research projects usually take longer to arrive at the final results than planned, and meanwhile some people might leave, others come in, and everyone forgets what they were exactly doing or thinking a few years ago. Even in this setting there are therefore good and bad ways to share data. Here are some of the main ways in which small research groups might share data, assessed within the context of the FAIR principles:</p>
<ul>
<li><strong>email</strong> - For small data sets (spreadsheets, for example) it might seem easy and quick to just email them as attachments. This is a bad idea for numerous reasons: 1) an email is uniquely <em>not</em> findable. Within a group of people you would have to refer to it as, for example, &quot;that email that I sent a few weeks ago&quot;, and this of course gets worse when other people start replying with other versions of data files, 2) there is no way to link to an email such that everyone has access to it, 3) email attachments have a size limit. <strong>Sending important files (such as data, manuscripts, or source code) by email within a research project should be discouraged under nearly all circumstances.</strong></li>
<li><strong>sneakernet</strong> - Large data sets (such as produced by an external DNA sequencing facility) are often shipped on external hard drives, and subsequently carried around the halls of a research institution, a practice referred to as &quot;<a href="https://en.wikipedia.org/wiki/Sneakernet" target="_blank">sneakernet</a>&quot;. This has obvious access problems (you somehow need to get access to the actual drive), but also a deeper &quot;findability&quot; problem: unless the team is disciplined in taking <a href="https://en.wikipedia.org/wiki/Checksum" target="_blank">checksums</a> of the data on the drive, there is no good way to tell whether the data have been changed or corrupted, and what the &quot;original&quot; data set was. <strong>Data from portable drives need to be copied onto a networked system, have checksums verified, and then shared using one of the methods below in such a way that the drive can subsequently be discarded without problems.</strong></li>
<li><strong>peer-to-peer</strong> - Numerous technologies exist for sending data directly &quot;<a href="https://en.wikipedia.org/wiki/Peer-to-peer" target="_blank">peer-to-peer</a>&quot;. An example of this that works for large blocks of data is <a href="http://wetransfer.com" target="_blank">WeTransfer</a>. This has similar problems as with email attachments: it is practically impossible to refer unambiguously to a specific transmission (and anyone that was not among the recipients in the first place will not be able to access it). This will make it hard to track <a href="../VERSIONING">versions</a>. In addition, peer-to-peer transfers are ephemeral (i.e. a service such as WeTransfer will remove the transferred file from its system after a certain amount of time). <strong>Taken together, peer-to-peer file transfer of important research data is not a good approach and should be discouraged.</strong></li>
<li><strong>HTTP servers</strong> - Online sharing systems such as <a href="http://dropbox.com" target="_blank">DropBox</a> and <a href="http://drive.google.com" target="_blank">Google Drive</a> have the great virtue that there is a unique <em>location</em> that all collaborators that have been granted access can refer to when talking about the data. For most of these systems, this location is a <a href="https://en.wikipedia.org/wiki/URL" target="_blank">HTTP URL</a>, which means that the uniqueness of the location is guaranteed by how the internet (i.e. <a href="https://en.wikipedia.org/wiki/Domain_Name_System" target="_blank">DNS</a>) works, that <a href="../WORKFLOWS">analytical workflows</a> can directly access the location (HTTP access is available in all common scripting languages and workflow systems), and that <a href="../SEMANTICS">semantic web</a> applications can make machine readable statements about the resource at the URL location. There are two main downsides: 1) given a URL for a file, it is not obvious how neighbouring files can be discovered (for example, there is no standard folder structure to navigate), 2) there is no standard way to track <a href="../VERSIONING">versions</a>. Different commercial systems (such as the aforementioned DropBox and Drive) have their own versioning systems, however, versions created by these systems are simply periodic snapshots without descriptive metadata (i.e. unlike a &quot;commit&quot; using version control systems such as <code>git</code> or <code>svn</code>). An open protocol that extends the functionality of HTTP, <a href="https://en.wikipedia.org/wiki/WebDAV" target="_blank">WebDAV</a>, forms the basis for how version control systems communicate with remote servers. Such version control systems are nearly perfect for FAIR data sharing, except for the bandwidth and storage limitations that these systems (may) have. Two optional extensions to git, <a href="https://git-lfs.github.com/" target="_blank">git-lfs</a> and <a href="https://git-annex.branchable.com/" target="_blank">git-annex</a>, have been developed to address these issues. <strong>Data sharing based on HTTP servers, especially when enhanced by open extensions for version and access control, is well worth considering when files meaningfully and notably change through time.</strong></li>
<li><strong>FTP servers</strong> - Data sharing using <a href="https://en.wikipedia.org/wiki/File_Transfer_Protocol" target="_blank">FTP</a> has the same advantage as HTTP in that it results in unambiguous locations based on DNS. In addition, it has the virtue that neighbouring locations on a server (e.g. the other files in a folder, and so on) can be navigated using a standard protocol supported by many tools. Bandwidth challenges are better addressed than in HTTP because downloads can be parallellised (trivially by opening different connections for different files, and by fetching multiple chunks of the same file in parallel). FTP is therefore one of the preferred methods for downloading large genomic data sets from <a href="https://www.ncbi.nlm.nih.gov/genome/doc/ftpfaq/" target="_blank">NCBI</a>. A downside is that there are no standards built on top of FTP for tracking file versions. Because FTP allows for navigating through file systems, this downside is commonly addressed by placing checksum files next to the data files. This allows for a quick check to see whether a remote file is the same version as a local copy without having to download the remote file. However, this is not sufficient for tracking entire file histories. <strong>Data sharing based on FTP servers is well worth considering for navigating static libraries of large files.</strong></li>
<li><strong>rsync servers</strong> - <a href="https://rsync.samba.org/" target="_blank">RSync</a> is an open source utility for synchronisation and incremental transfers between file systems. The advantage of this system is that libraries of large files can be kept in sync between systems, so if you are keeping sets of files where occasionally some change (such as a library of reference genomes that go through releases), this system is more efficient with bandwidth than FTP, which is why NCBI prefers it over FTP. Using RSync means that changed files will be updated (which is why a number of backup systems are based on it) but it is not a versioning system. <strong>Data sharing using RSync is appropriate for synchronising file structures across systems, assuming that the changes in the file structures are managed by some other systems (such as release notes).</strong> </li>
</ul>
<p>The above solutions are well within reach of every researcher: the server technologies that are recommended are either already installed on some operating systems, or are freely available as very popular, well-documented open source software with large user communities. For larger institutions that have the ICT staffing resources to maintain more advanced data sharing infrastructures, two open source systems are worth mentioning within the context of bioinformatics applications: <a href="https://irods.org/" target="_blank">iRODS</a> and <a href="https://www.globus.org/" target="_blank">Globus</a>. What these systems add is that they can track (meta)data, and thus file versions,  across distributed architectures and protocols. For example, such systems can figure out that the raw data file you are trying to fetch exists as a local copy near you so as to use bandwidth more efficiently, or allow for the discovery of related files by custom metadata that researchers attach to their project files, integrate file management in data processing workflows, and a lot more. The downside of these systems lies in the complexity of installing, configuring and maintaining these systems.</p>
<h2 id="publishing-result-data">Publishing result data</h2>
<p>The final outcomes of a data-intensive research project in terms of data are referred to as &quot;result data&quot;. For example, for a genome re-sequencing project, these might be the specific gene variants detected in the individuals that were sequenced (e.g. see <a href="../VERSIONING#v3">this pyramid</a>). These are the data that scientific conclusions - such as presented in a scholarly publication - will probably be most intimately based on. (However, these cannot be decoupled from  data nearer to the base of the pyramid, so developing an approach that allows you to &quot;drill down&quot; from the top to these lower levels is vital.) and The pressures to share result data come from more sides than in the case of raw or intermediate data:</p>
<ol>
<li>Funding agencies more and more require that result data are shared. Research projects funded by such agencies probably need to submit a <a href="../DATA_MANAGEMENT">data management plan</a> for approval to the agency that describes how this will be handled. (The NIH, the US funding agency for medical research, has published a list of <a href="https://grants.nih.gov/grants/policy/data_sharing/data_sharing_faqs.htm" target="_blank">frequently asked questions</a> surrounding data sharing that researchers might ask a funding agency.)</li>
<li>Many journals require that data discussed in a paper are deposited in a data repository such that the paper refers to a public, clearly identifiable, data record. (In addition to policies of individual journals, community initiatives to establish author guidelines to this end are being developed, see [<a href="../REFERENCES#Nosek2015">Nosek2015</a>].)</li>
<li>A publication based on open, reusable data is more likely to be cited (e.g. by 9% in the case of microarray data [<a href="../REFERENCES#Piwowar2013">Piwowar2013</a>]), so there is also an aspect of enlightened self interest to depositing result data. Along this vein, a growing trend is towards &quot;<a href="../REFERENCES#Chavan2011">data publications</a>&quot;, where the paper is mostly just an advertisement for a re-usable data set.</li>
<li>Data sets themselves might be citable, which is a vision that is being advanced, for example, by the <a href="https://www.force11.org/group/joint-declaration-data-citation-principles-final" target="_blank">Joint Declaration of Data Citation Principles</a> and by infrastructural initiatives such as <a href="https://www.datacite.org/" target="_blank">DataCite</a> and the <a href="http://wokinfo.com/products_tools/multidisciplinary/dci/" target="_blank">Data Citation Index</a>.</li>
</ol>
<p>These different pressures have created a need for online data repositories, and in response a bewildering array of choices has arisen. Broadly speaking, these can be subdivided in generic data repositories that will accept many different data types but will not process (e.g. validate, index, visualise) these in great detail, and domain-specific repositories that do a lot more with the data. The choices are discussed below.</p>
<h3 id="generic-data-repositories">Generic data repositories</h3>
<p>In what concerns platforms to share various types of data, a researcher has access to a reasonable amount of choice. At present (August 2017), we invite you to explore a number of repositories. Often they are complemented with a series of services that offer some specific advantages, such as linking to publications, providing licensing options, offering digital object identifiers (DOI). They differ quite considerably in their policies, level of curation and organization methods. Specificity also occurs in the case of repositiories orientated towards certain communities of users. Such is the case of digital libraries, for example.</p>
<ul>
<li><p><strong>Dryad Digital Repository</strong> &quot;The Dryad Digital Repository is a curated resource that makes the data underlying scientific publications discoverable, freely reusable, and citable. Dryad provides a general-purpose home for a wide diversity of datatypes.&quot; <a href="http://datadryad.org/" target="_blank">http://datadryad.org/</a></p>
</li>
<li><p><strong>FigShare</strong> &quot;As governments and funders of research see the benefit of open content, the creation of recommendations, mandates and enforcement of mandates are coming thick and fast. figshare has always led the way in enabling academics, publishers and institutions to easily adhere to these principles in the most intuitive and efficient manner.&quot; Mark Hahnel, Founder and CEO, figshare <a href="https://figshare.com/" target="_blank">https://figshare.com/</a></p>
</li>
<li><p><strong>Zenodo</strong> &quot;Zenodo helps researchers receive credit by making the research results citable and through OpenAIRE integrates them into existing reporting lines to funding agencies like the European Commission. Citation information is also passed to DataCite and onto the scholarly aggregators.&quot; <a href="https://zenodo.org/" target="_blank">https://zenodo.org/</a></p>
</li>
<li><p><strong>Dataverse</strong> &quot;Dataverse is an open source web application to share, preserve, cite, explore, and analyze research data. It facilitates making data available to others, and allows you to replicate others&apos; work more easily. Researchers, data authors, publishers, data distributors, and affiliated institutions all receive academic credit and web visibility.&quot; <a href="https://dataverse.org/" target="_blank">https://dataverse.org/</a></p>
</li>
<li><p><strong>EUDAT</strong> &quot;EUDAT offers heterogeneous research data management services and storage resources, supporting multiple research communities as well as individuals, through a geographically distributed, resilient network distributed across 15 European nations and data is stored alongside some of Europe&apos;s most powerful supercomputers.&quot; <a href="https://eudat.eu/" target="_blank">https://eudat.eu/</a></p>
</li>
<li><p><strong>Mendeley Data</strong> &quot;Mendeley Data is a secure cloud-based repository where you can store your data, ensuring it is easy to share, access and cite, wherever you are.&quot; <a href="https://data.mendeley.com/" target="_blank">https://data.mendeley.com/</a></p>
</li>
</ul>
<h3 id="domain-specific-repositories">Domain-specific repositories</h3>
<p>In addition to the generic data repositories listed above, a very large number of domain-specific data repositories and databases exists. Such repositories accept only a limited number of data types - for example, DNA sequences - that need to be provided in specific formats and with specific metadata. This means that, probably, not all different data and results generated by a research project can be uploaded to the same repository, and that uploading is sometimes cumbersome and complicated. On the other hand domain-specific repositories can provide more services tailored to a specific data type (such as BLAST searching of uploaded DNA sequences), and perhaps do things with the data such as synthesizing it in a meta-analysis or a computed consensus. Nature publishes an <a href="https://www.nature.com/sdata/policies/repositories" target="_blank">online list</a> of a variety of data repositories, while <a href="http://www.re3data.org/" target="_blank">re3data</a> provides a searchable database of repositories.</p>
<h2 id="licensing-attribution-and-openness-in-data-repositories">Licensing, Attribution and Openness in Data Repositories</h2>
<p>Data made available via open repositories enables a series of attractive opportunities for researchers. First, the opportunity for attaching an attribution to the work of capturing, storing and curating data. Credit is assigned to the depositor following well established and understood rules, in general this is ensure by assigning one of the variants of the <a href="https://creativecommons.org/licenses/" target="_blank">Creative Commons licensing scheme</a>. (Refer to <a href="../SCIENTIFIC_SOFTWARE#rights-and-licenses">the equivalent session on this topic for software development</a> to see what is done with source code.)</p>
<p>Data sharing in this way also enables reuse and, in particular, reprocessing. This exposure of data to other resarchers, students and the public is gaining popularity. It is obvious that others may find more from the same data by using different tools with different parameter settings and options, and it is interesting to observe that it can be used preserving attribution. On the other hand, the opportunities that arise when combining heterogeneous datasets may allow for important steps that would be much more difficult otherwise. </p>
<h2 id="expected-outcomes">Expected outcomes</h2>
<p>In this section we have discussed the different infrastructural solutions for sharing research data and how these relate to the developing principles for data sharing. You should now be able to:</p>
<ul>
<li>Explain what the terms of the <strong>FAIR</strong> acronym stand for in relation to data sharing</li>
<li>Explain the difference between sharing raw and intermediate data, and result data</li>
<li>Assess the advantages and drawbacks of different data sharing approaches within teams</li>
<li>Be able to locate the appropriate domain-specific repository for a given, common research data type (e.g. DNA sequence)</li>
<li>Be able to locate generic repositories for opaque research data and weigh their advantages and drawbacks against domain-specific ones</li>
</ul>

                                
                                </section>
                            
    </div>
    <div class="search-results">
        <div class="has-results">
            
            <h1 class="search-results-title"><span class='search-results-count'></span> results matching "<span class='search-query'></span>"</h1>
            <ul class="search-results-list"></ul>
            
        </div>
        <div class="no-results">
            
            <h1 class="search-results-title">No results matching "<span class='search-query'></span>"</h1>
            
        </div>
    </div>
</div>

                        </div>
                    </div>
                
            </div>

            
                
                <a href="../DATA_MANAGEMENT/" class="navigation navigation-prev " aria-label="Previous page: How to manage data and plan for it">
                    <i class="fa fa-angle-left"></i>
                </a>
                
                
                <a href="../WORKFLOWS/" class="navigation navigation-next " aria-label="Next page: How to do analyses in a workflow-oriented manner">
                    <i class="fa fa-angle-right"></i>
                </a>
                
            
        
    </div>

    <script>
        var gitbook = gitbook || [];
        gitbook.push(function() {
            gitbook.page.hasChanged({"page":{"title":"How to share research data fairly and sensibly","level":"3.3","depth":1,"next":{"title":"How to do analyses in a workflow-oriented manner","level":"4.1","depth":1,"path":"WORKFLOWS/README.md","ref":"WORKFLOWS/README.md","articles":[]},"previous":{"title":"How to manage data and plan for it","level":"3.2","depth":1,"path":"DATA_MANAGEMENT/README.md","ref":"DATA_MANAGEMENT/README.md","articles":[]},"dir":"ltr"},"config":{"plugins":["mathjax","quizzes","codeblock-disable-glossary","image-captions","livereload"],"root":"./docs","styles":{"website":"styles/website.css","pdf":"styles/pdf.css","epub":"styles/epub.css","mobi":"styles/mobi.css","ebook":"styles/ebook.css","print":"styles/print.css"},"pluginsConfig":{"livereload":{},"search":{},"autocover":{"title":"Open Science, Open Data, Open Source","author":"Pedro L. Fernandes and Rutger A. Vos","font":{"size":null,"family":"Impact","color":"#FFF"},"size":{"w":1800,"h":2360},"background":{"color":"#006600"}},"lunr":{"maxIndexSize":1000000,"ignoreSpecialCharacters":false},"codeblock-disable-glossary":{},"fontsettings":{"theme":"white","family":"sans","size":2},"quizzes":{},"highlight":{},"mathjax":{"forceSVG":false,"version":"2.6-latest"},"sharing":{"facebook":true,"twitter":true,"google":false,"weibo":false,"instapaper":false,"vk":false,"all":["facebook","google","twitter","weibo","instapaper"]},"theme-default":{"styles":{"website":"styles/website.css","pdf":"styles/pdf.css","epub":"styles/epub.css","mobi":"styles/mobi.css","ebook":"styles/ebook.css","print":"styles/print.css"},"showLevel":false},"image-captions":{"variable_name":"_pictures"}},"theme":"default","pdf":{"pageNumbers":true,"fontSize":12,"fontFamily":"Arial","paperSize":"a4","chapterMark":"pagebreak","pageBreaksBefore":"/","margin":{"right":62,"left":62,"top":56,"bottom":56}},"structure":{"langs":"LANGS.md","readme":"README.md","glossary":"GLOSSARY.md","summary":"SUMMARY.md"},"variables":{"_pictures":[{"backlink":"LITERATURE_STUDY/index.html#fig2.1.1","level":"2.1","list_caption":"Figure: Randall Munroe on open access publishing","alt":"Randall Munroe on open access publishing","nro":1,"url":"LS2.jpg","index":1,"caption_template":"Figure: _CAPTION_","label":"Randall Munroe on open access publishing","attributes":{},"skip":false,"key":"2.1.1"},{"backlink":"PUBLISHING/index.html#fig2.4.1","level":"2.4","list_caption":"Figure: Preprints","alt":"Preprints","nro":2,"url":"https://img.youtube.com/vi/2zMgY8Dx9co/0.jpg","index":1,"caption_template":"Figure: _CAPTION_","label":"Preprints","attributes":{},"skip":false,"key":"2.4.1"}]},"gitbook":"*"},"file":{"path":"DATA_SHARING/README.md","mtime":"2017-09-03T11:57:10.000Z","type":"markdown"},"gitbook":{"version":"3.2.3","time":"2017-09-03T13:49:38.545Z"},"basePath":"..","book":{"language":""}});
        });
    </script>
</div>

        
    <script src="../gitbook/gitbook.js"></script>
    <script src="../gitbook/theme.js"></script>
    
        
        <script src="https://cdn.mathjax.org/mathjax/2.6-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-mathjax/plugin.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-quizzes/quizzes.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-codeblock-disable-glossary/disable-glossary.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-livereload/plugin.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-search/search-engine.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-search/search.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-lunr/lunr.min.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-lunr/search-lunr.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-sharing/buttons.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-fontsettings/fontsettings.js"></script>
        
    

    </body>
</html>

